### PR TITLE
Add AKS Cluster v2 Support

### LINKED ISSUE #1: Improve example with version 2.0
🐞 **Problem**

Add examples for AKS using `version = "~> 2.0"`

🎯 **Goal**

Improve examples with modern version of `azurerm` provider

💡 **Possible solutions**

Change `agent_pool_profile` by `default_node_pool` and `linux_profile`

### PR DESCRIPTION
**Pull request by imjoseangel**
Fixes #1 

I have added a v2 directory and keep the previous version. IMHO I think it could be better to replace the oldest with the new requirements. Waiting for your feedback.

Thank you

### CODE REVIEW

**Comment by imjoseangel** [comment]
@im2nguyen Could you please have a look? Thank you!

File: version2/aks-cluster.tf [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  addon_profile {
+      oms_agent {
+      enabled     = false
+      }
+  }
+
+  network_profile {
+    load_balancer_sku = "Standard"
+    network_plugin = "kubenet"
+  }
```
Comment (im2nguyen) [inline version2/aks-cluster.tf]: I believe these (`addon_profile` and `network_profile`) are the default for the AKS cluster, let's omit them?

File: version2/aks-cluster.tf [hunk 2]
```diff
@@ -0,0 +1,64 @@
+resource "random_pet" "prefix" {}
+
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  client_id       = var.client_id
+  client_secret   = var.client_secret
+  tenant_id       = var.tenant_id
```
Comment (im2nguyen) [inline version2/aks-cluster.tf]: Let's get rid of these. Provider auth should be done using environment variables or using the Azure CLI profile, this is so credentials aren't accidentally committed to version control

File: version2/aks-cluster.tf [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    os_disk_size_gb = 30
+  }
+
+  linux_profile {
+      admin_username = "k8s"
+
+      ssh_key {
+          key_data = file(var.ssh_public_key)
+      }
+  }
```
Comment (im2nguyen) [inline version2/aks-cluster.tf]: Planning on removing this too, this repo is meant to serve as a quick way to get started provisioning an AKS cluster

Users can then customize their cluster according to their use case by referencing the [`azurerm` Registry](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster) or referencing the [examples](https://github.com/terraform-providers/terraform-provider-azurerm/tree/master/examples/kubernetes) listed on the `terraform-provider-azurerm` GitHub page

I'll add links to the `terraform-provider-azurerm` GitHub page examples on the [Learn AKS guide](https://learn.hashicorp.com/tutorials/terraform/aks?in=terraform/kubernetes)

File: version2/terraform.tfvars [hunk 4]
```diff
@@ -0,0 +1,6 @@
+appId    = "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"
+password = "********-****-****-****-************"
+tenant_id = "bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb"
+subscription_id = "cccccccc-cccc-cccc-cccc-cccccccccccc"
+client_id = "dddddddd-dddd-dddd-dddd-dddddddddddd"
+client_secret = "********-****-****-****-************"
```
Comment (im2nguyen) [inline version2/terraform.tfvars]: Remove these from config

**Review by im2nguyen — COMMENTED** [review]
This looks like a good start! Can you make these changes to the main files (remove the `version2`)?

The purpose of this repo is to serve as a quick start for users. Once they complete this, they can reference the [`terraform-provider-azurerm` examples repo] or the [Terraform Registry](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster) to configure AKS clusters to their specific use case.

File: version2/aks-cluster.tf (hunk 1 above)
Comment (imjoseangel) [reply version2/aks-cluster.tf]: Changed Thanks

**Comment by imjoseangel** [comment]
Thank you @im2nguyen 

All done!

**Comment by im2nguyen** [comment]
This looks good! Will merge after updating the guide. Thanks!!

**Comment by imjoseangel** [comment]
> This looks good! Will merge after updating the guide. Thanks!!

Let me know if you need help on that part. Thanks

**Comment by im2nguyen** [comment]
Guide has been updated, thanks again @imjoseangel !

**Pull request merged**